### PR TITLE
Nested rows will be displayed properly and no error will be thrown while removing last table element #7548

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

This PR is fixing 2 problems:
- improper displaying nested rows at start - described with #7548 (some CRUD actions still doesn't work)
- removing last row has thrown error

For the first problem, there has been provided a workaround. Since Handsontable 8.0.0 there is a problem with proper handling of changes from external sources, ie. modifying table's data through changing data feeding the table, provided by wrapper. [Very simple solution](https://github.com/handsontable/vue-handsontable-official/blob/12a2d8cb4b17db6e2b7dd97489a6d91b3d374b06/src/HotTable.vue#L31-L34) which has worked until Handsontable 8 release won't be appropriate anymore. We can't just call an `updateSettings` with `data` key as it resets sequence of indexes for **every change**. Code [provided here](handsontable/vue-handsontable-official@886703c#diff-868ba4f09336612013cd40b69c3fc77b3724064bbe38ef887d4ab83100525608R42) isn't perfect. However, the problem seems to be out of the scope of the task.

Some other improvements could be done after developing https://github.com/handsontable/handsontable/issues/7263.

### How has this been tested?

Manual tests in application bundled by Parcel (Handsontable with nested rows and simpler data).

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #7548 

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [x] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Review by aninde — CHANGES_REQUESTED** [review]
- removing the last row has thrown an error
Do you mean deleting the last parent? For me, it wasn't fixed. 
Check on the demo https://jsfiddle.net/aninde/ndp4tk9y/
Maybe adding Vue-Handsontable it's not sufficient to present it.

removing-nestedparent

If you mean removing the last row (child of nested row) it works correctly, but it can't be undone. I suppose it is off the scop of this PR.

I miss the unit test for this case.

**Comment by wszymanski** [comment]
> * removing the last row has thrown an error
>   Do you mean deleting the last parent? For me, it wasn't fixed.

Thanks for the question. It's not about `NestedRows` plugin. I wrote `(some CRUD actions still doesn't work)` in context of the plugin.

> I miss the unit test for this case.

There is extra test check for [the mentioned fix](https://github.com/handsontable/handsontable/pull/7720/files#diff-f4caf4afce92b093ab63aea51662ddc8fcba8c5750b599b425515e7f9b23bb57R209-R215).

**Comment by aninde** [comment]
> > I miss the unit test for this case.
> 
> There is an extra test check for [the mentioned fix](https://github.com/handsontable/handsontable/pull/7720/files#diff-f4caf4afce92b093ab63aea51662ddc8fcba8c5750b599b425515e7f9b23bb57R209-R215).

Yes, this change is in commits on this PR but, I don't have it in the code on the version built from this PR.
I miss all the changes from commit 6637391c1098fb2198578c92ecefb3d4c0f4981e https://github.com/handsontable/handsontable/pull/7720/files#diff-f4caf4afce92b093ab63aea51662ddc8fcba8c5750b599b425515e7f9b23bb57R209-R215.

**Comment by wszymanski** [comment]
> > > I miss the unit test for this case.
> > 
> > 
> > There is an extra test check for [the mentioned fix](https://github.com/handsontable/handsontable/pull/7720/files#diff-f4caf4afce92b093ab63aea51662ddc8fcba8c5750b599b425515e7f9b23bb57R209-R215).
> 
> Yes, this change is in commits on this PR but, I don't have it in the code on the version built from this PR.

Hmm? This test exist in final version of code.

**Comment by aninde** [comment]
> Hmm? This test exist in final version of code.

True, my fault.

File: wrappers/vue/test/hotTable.spec.ts [hunk 1]
```diff
@@ -206,6 +206,13 @@ describe('Updating the Handsontable settings', () => {
 
     expect(testWrapper.vm.$children[0].hotInstance.getData()).toEqual([[22, 32, 42]]);
     expect(newHotSettings).toBe(null);
+
+    testWrapper.vm.removeRow();
```
Comment (aninde) [inline wrappers/vue/test/hotTable.spec.ts:210]: I don't understand the reliability of this test.
Function `removeRows` corresponds to
line 167 
 ```
 removeRow: function () {
           this.data.pop ()
         }

```
Why do we consider decreasing the dataset as deleting a row?

Comment (wszymanski) [reply wrappers/vue/test/hotTable.spec.ts:210]: We consider. However, it doesn't work good for nested rows, as I mentioned (bug). This test checks refers to removing the last row.

**Review by aninde — APPROVED** [review]
👍

**Review by budnix — APPROVED** [review]
LGTM 👍

**Pull request merged**